src/middlewared/middlewared/plugins/iscsi_/extents.py

```diff
@@ -105,11 +105,11 @@ async def do_create(self, data):
         `ro` when set to true prevents the initiator from writing to this LUN.
         """
         verrors = ValidationErrors()
-        await self.validate(data)
+        await self.middleware.run_in_thread(self.validate, data)
```

Comment:
This is perfectly working code, but here is how we do it everywhere else in order to be agnostic of whether called function is synchronous or asynchronous:
```suggestion
        await self.middleware.call('iscsi.extent.validate', data)
```

Reply:
Thx @themylogin -- is your point more about performing a `call` than a `run_in_thread` ?

Reply:
@rick-mesta yes